### ISSUE TITLE
Forecast error bands seem potentially implausible

### ISSUE THREAD

**Issue by dajmcdon** [epiforecasts/EpiNow2#434]
I noticed this possible issue during a recent talk on epinow2/epinowcast. The credible intervals for the forecast of Rt seem strange to me. One would think that their width would grow, similar to the partial data CIs. See, for example, Panel C on the [README](https://epiforecasts.io/EpiNow2/dev/#epinow). I can imagine a few potential reasons for this, due to missing uncertainty somewhere. 

One possibility is that the specification for the GP doesn't extrapolate correctly. It's possible that directly using a state-space system instead could both (a) fix the CIs and (b) speed up fitting significantly (depending on the STAN implementation).

**Comment by sbfnk** [comment]
This behaviour is controlled by the `future` argument in [rt_opts](https://epiforecasts.io/EpiNow2/reference/rt_opts.html) - its default is "latest", which keeps the reproduction number constant beyond the latest estimate based on partial data. We decided to use this as a default because we found that correctly extrapolating the GP beyond the last data point led to CIs on the infection scale that were too wide to be useful for forecasting.

Here's the plot from the README generated with `future = "project"`:
<img width="400" alt="image" src="https://github.com/epiforecasts/EpiNow2/assets/1156307/1ae51608-0fae-43f0-8296-d544fa6c35af">

We experimented with other approaches to modelling future behaviour of Rt in the [EpiSoon](https://github.com/epiforecasts/EpiSoon/) package but didn't end up taking this further.

Perhaps I'm misunderstanding but I don't see how a different formulation of the GP would speed up fitting given that this behaviour is only very weakly informed by the data.
